src/i2c.rs

```diff
@@ -1,14 +1,43 @@
 //! Helper functions for the u16 word based I²C communication.
 
 use crate::crc8;
+use core::convert::TryFrom;
 use embedded_hal::blocking::i2c;
 
 /// All possible errors in this crate
 #[derive(Debug, PartialEq, Copy, Clone)]
 pub enum Error<I2cWrite: i2c::Write, I2cRead: i2c::Read> {
     I2cWrite(I2cWrite::Error),
     I2cRead(I2cRead::Error),
-    Crc,
+    CrcError,
+    InvalidBufferSize,
+}
+
+impl<W: i2c::Write, R: i2c::Read> From<crc8::Error> for Error<W, R> {
+    fn from(err: crc8::Error) -> Error<W, R> {
+        match err {
+            crc8::Error::CrcError => Error::CrcError,
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Copy, Clone)]
+pub enum ConversionError {
+    InvalidBufferSize,
+}
+
+#[derive(Debug)]
+pub struct I2CBuffer<'a>(pub &'a mut [u8]);
```

Comment:
Can you add some documentation that the buffer must hold a multiple of three bytes?

Reply:
Also I think the member shouldn't be `pub`, because with `pub` it can be constructed without the `TryFrom`, no?

Reply:
I set the member to `pub(crate)`, because the `crc8` needs to be able to see it.